File: PythonParser/AssignmentStatement.py
__author__ = 'ravipatel'

from Memory import store
from Statement import Statement


class Assignment_Statement(Statement):
    """

    """
    def __init__(self,var1, expr1):
        if (var1 is None):
            raise ValueError("Error for var1 in assignment_Statement class")
        self.var1 = var1
        if (expr1 is None):
            raise ValueError("Error for expr1 in assignment_statement class")
        self.expr1 = expr1

    def execute(self):
        store(self.var1.getVar(), self.expr1.evaluate())

File: JavaParser/Assignment_Statement.java

public class Assignment_Statement implements Statement
{
	private Id var1;
	
	private Expression expr1;
	
	public Assignment_Statement()
	{	}
	
	public Assignment_Statement(Id var1, Expression expr1)
	{
		if (var1 == null)
			throw new IllegalArgumentException("var1 error/AssignmentStatement");
		if (expr1 == null)
			throw new IllegalArgumentException("expr1 error /AssignmentStatement");
		
		this.var1 = var1;
		this.expr1 = expr1;
	}
	
	@Override
	public void execute()
	{
		//method to store var and evaluate for a method in memory
		Memory.store(var1.getChar(), expr1.evaluate());
	}
}
